client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/internal/download/DownloaderTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.internal.download;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import io.micronaut.runtime.server.EmbeddedServer;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import uk.gov.gchq.palisade.client.internal.dft.DefaultQueryResponse.EmittedResource;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.http.HttpClient;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+@MicronautTest
+class DownloaderTest {
+
+    private static final String BASE_URL = "http://localhost:%d"; // needs port added before use
+    private static final String ENDPOINT = "/data/read/chunked";
```

Comment:
In use, the URL returned by the resource points directly to the data-service, and the endpoint path should then be "/read/chunked"

Reply:
This has been changed